protocols/imap/src/main/java/org/apache/james/imap/processor/GetAnnotationProcessor.java

```diff
@@ -0,0 +1,165 @@
+                         List<MailboxAnnotation> mailboxAnnotations, long maxsize, Optional<Integer> maximumOversizedSize) {
+        if (maximumOversizedSize.isPresent()) {
+            responder.respond(new AnnotationResponse(mailboxName, filterItemsBySize(responder, mailboxName, mailboxAnnotations, maxsize)));
+            okComplete(command, tag, ResponseCode.longestMetadataEntry(maximumOversizedSize.get()), responder);
+        } else {
+            responder.respond(new AnnotationResponse(mailboxName, mailboxAnnotations));
+            okComplete(command, tag, responder);
+        }
+    }
+
+    private Optional<Integer> getMaxSizeValue(List<MailboxAnnotation> mailboxAnnotations, long maxsize) {
+        if (maxsize > 0) {
+            return getMaxSizeOfOversizedItems(mailboxAnnotations, maxsize);
+        }
+        return Optional.absent();
+    }
+
+    private List<MailboxAnnotation> filterItemsBySize(Responder responder, String mailboxName, List<MailboxAnnotation> mailboxAnnotations, final long maxsize) {
+        Predicate<MailboxAnnotation> lowerPredicate = new Predicate<MailboxAnnotation>() {
+            @Override
+            public boolean apply(MailboxAnnotation input) {
+                return (input.size() <= maxsize);
+            }
+        };
+
+        return FluentIterable.from(mailboxAnnotations).filter(lowerPredicate).toList();
+    }
+
+    private List<MailboxAnnotation> getMailboxAnnotations(ImapSession session, Set<MailboxAnnotationKey> keys, GetAnnotationRequest.Depth depth, MailboxPath mailboxPath) throws MailboxException {
+        MailboxSession mailboxSession = ImapSessionUtils.getMailboxSession(session);
+        switch (depth) {
+            case ZERO:
+                return getMailboxAnnotationsWithDepthZero(keys, mailboxPath, mailboxSession);
+            case ONE:
+                return getMailboxManager().getAnnotationsByKeysWithOneDepth(mailboxPath, mailboxSession, keys);
+            case INFINITY:
+                return getMailboxManager().getAnnotationsByKeysWithAllDepth(mailboxPath, mailboxSession, keys);
+            default:
+                throw new NotImplementedException();
+        }
+    }
+
+    private List<MailboxAnnotation> getMailboxAnnotationsWithDepthZero(Set<MailboxAnnotationKey> keys, MailboxPath mailboxPath, MailboxSession mailboxSession) throws MailboxException {
+        if (keys.isEmpty()) {
+            return getMailboxManager().getAllAnnotations(mailboxPath, mailboxSession);
+        } else {
+            return getMailboxManager().getAnnotationsByKeys(mailboxPath, mailboxSession, keys);
+        }
+    }
+
+    private Optional<Integer> getMaxSizeOfOversizedItems(List<MailboxAnnotation> mailboxAnnotations, final long maxsize) {
+        Predicate<MailboxAnnotation> filterOverSizedAnnotation = new Predicate<MailboxAnnotation>() {
+            @Override
+            public boolean apply(MailboxAnnotation input) {
+                return (input.size() >= maxsize);
```

Comment:
You need some consistency on what happen with annotation whose size is the limit.

From line 106 : These annotation are still returned.

From line 139: They are concidered over sized.

They can not be both. Do you have test cases on this hedge case ? What does the RFC says ?